src/components/DataSetSelect/DataSetSelect.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { Select, Popover } from 'antd';
+import { canReadTypes } from '../../utils/PermissionsUtils';
+import { useSelector, useDispatch } from '../../utils/ReduxUtils';
+import { list, selectAllDataSets, DataSet } from '../../modules/datasets';
+
+type Props = {
+  onDataSetSelected: (ids: number[]) => void;
+  style?: React.CSSProperties;
+  disabled?: boolean;
+  multiple?: boolean;
+  selectedDataSetIds?: number[];
+};
+
+const DataSetSelect = ({
+  onDataSetSelected,
+  style = { width: '200px' },
+  disabled = false,
+  multiple = false,
+  selectedDataSetIds,
+}: Props) => {
+  const dispatch = useDispatch();
+  const datasets = useSelector(selectAllDataSets);
+  const [searchResults, setSearchResults] = useState([] as DataSet[]);
+  const [currentSelection, setCurrentSelection] = useState([] as number[]);
+  const [query, setQuery] = useState('');
+
+  const setSelectedValue = (ids?: number | number[]) => {
+    if (!ids) {
+      setCurrentSelection([]);
+      onDataSetSelected([]);
+    } else if (multiple) {
+      setCurrentSelection(ids as number[]);
+      onDataSetSelected(ids as number[]);
+    } else {
+      setCurrentSelection([ids as number]);
+      onDataSetSelected([ids as number]);
+    }
+    setQuery('');
+  };
+
+  useEffect(() => {
+    const doSearch = async () => {
```

Comment:
is this function really needed?

Reply:
huh i guess it was async but its not anymore, will fix